src/metrics/loc.rs

```diff
@@ -142,6 +140,47 @@ fn init(node: &Node, stats: &mut Stats, is_func_space: bool, is_unit: bool) -> (
     (start, end)
 }
 
+#[inline(always)]
+// Discriminates among the comments that are *after* a code line and
+// the ones that are on an independent line.
+// This difference is necessary in order to avoid having
+// a wrong count for the blank metric.
+fn add_cloc_lines(stats: &mut Stats, start: usize, end: usize) {
+    let comment_diff = end - start;
+    let is_comment_afer_code_line = stats.lines.contains(&start);
+    if is_comment_afer_code_line && comment_diff == 0 {
+        // A comment is *entirely* next to a code line
+        stats.code_comment_lines += 1;
+    } else if is_comment_afer_code_line && comment_diff > 0 {
+        // A block comment that starts next to a code line and ends on
+        // independent lines.
+        stats.code_comment_lines += 1;
+        stats.only_comment_lines += comment_diff;
+    } else {
+        // A comment on an independent line AND
+        // a block comment on independent lines OR
+        // a comment *before* a code line
+        stats.only_comment_lines += (end - start) + 1;
+        // Save line end of a comment to check whether
+        // a comment *before* a code line is considered
+        stats.comment_line_end = Some(end);
+    }
+}
+
+#[inline(always)]
+// Detects the comments that are on a code line but *before* the code part.
+// This difference is necessary in order to avoid having
+// a wrong count for the blank metric.
+fn cloc_before_code_line(stats: &mut Stats, start_code_line: usize) {
+    if let Some(end) = stats.comment_line_end {
+        if !stats.lines.contains(&start_code_line) && end == start_code_line {
```

Comment:
Move the `end == start_code_line` condition before `!stats.lines.contains(&start_code_line)`, since it is faster to evaluate